### PR TITLE
add Tests

### PR DESCRIPTION
**Pull request by SomethingWittyWithATechAspect**
Test for Start menu are not running - dont know if they worked beforehand

walls don't seem to be fixed - Collsion Test are failing

### CODE REVIEW

**Comment by SomethingWittyWithATechAspect** [comment]
also didnt found where menutest is, menu test is most likely in the wrong dir
expect im blind

File: Assets/Editor/Tests/CameraTest.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        Assert.GreaterOrEqual(altUnityDriver.FindObject(By.TAG, "MainCamera").getWorldPosition().y, startY);
+    }
+
+    //At the end of the test closes the connection with the socket
+    [OneTimeTearDown]
+    public void TearDown()
+    {
+        altUnityDriver.Stop();
+    }
+}
```
Comment (spirit126) [inline Assets/Editor/Tests/CameraTest.cs:42]: no newline at end of file

File: Assets/Editor/Tests/CollisionTest.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    [Test]
+    public void TestCollisionDown()
+    {
+        float start = altUnityDriver.FindObject(By.NAME, "WallDown").y;
+        altUnityDriver.PressKey(AltUnityKeyCode.S, 1F, 2F, true);
+        Assert.GreaterOrEqual(altUnityDriver.FindObject(By.NAME, "Player").y, start);
+    }
+}
```
Comment (spirit126) [inline Assets/Editor/Tests/CollisionTest.cs:66]: no newline

File: Assets/Editor/Tests/ShooterTest.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        try
+        {
+            altUnityDriver.FindObject(By.NAME, "Enemy");
+            Assert.IsTrue(false);
+        }
+        catch (Altom.AltUnityDriver.NotFoundException e)
+        {
+            Assert.IsTrue(true);
+            Assert.IsTrue(7.Equals( altUnityDriver.FindObjects(By.TAG, "Projectile").Count));
+        }
```
Comment (spirit126) [inline Assets/Editor/Tests/ShooterTest.cs:54]: why not use Assert.Equals

File: Assets/Editor/Tests/ShooterTest.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    [Test]
+    public void TestShootWall()
+    {
+        var wall = altUnityDriver.FindObject(By.TAG, "Wall");
+        Assert.NotNull(wall);
+        altUnityDriver.Click(new AltUnityVector2(wall.x, wall.y), 10, 0.3F, true);
+        Assert.IsTrue(2.Equals( altUnityDriver.FindObjects(By.TAG, "Projectile").Count));
```
Comment (spirit126) [inline Assets/Editor/Tests/ShooterTest.cs:63]: why not use Assert.Equals

File: Assets/Editor/Tests/ShooterTest.cs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    
+    //At the end of the test closes the connection with the socket
+    [OneTimeTearDown]
+    public void TearDown()
+    {
+        altUnityDriver.Stop();
+    }
+}
```
Comment (spirit126) [inline Assets/Editor/Tests/ShooterTest.cs:73]: no newline

File: Assets/Editor/Tests/CameraTest.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
+using NUnit.Framework;
+using Altom.AltUnityDriver;
+using UnityEngine;
+using UnityEngine.TestTools;
+using NUnit.Framework;
+
+public class CameraTest : MonoBehaviour
+{
+    private AltUnityDriver altUnityDriver = new AltUnityDriver();
+    private string SceneName = "Main";
```
Comment (spirit126) [inline Assets/Editor/Tests/CameraTest.cs:10]: altUnityDriver -> _altUnityDriver, and can be made readonly
SceneName: naming convention and can be made constant

File: Assets/Editor/Tests/CollisionTest.cs [hunk 7]
```diff
@@ -0,0 +1,66 @@
+using NUnit.Framework;
+using Altom.AltUnityDriver;
+using UnityEngine;
+
+[TestFixture]
+public class CollisionTest : MonoBehaviour
+{
+    private AltUnityDriver altUnityDriver = new AltUnityDriver();
+
```
Comment (spirit126) [inline Assets/Editor/Tests/CollisionTest.cs:9]: same as above

File: Assets/Editor/Tests/ShooterTest.cs [hunk 8]
```diff
@@ -0,0 +1,73 @@
+using System;
+using NUnit.Framework;
+using Altom.AltUnityDriver;
+using UnityEngine;
+
+public class ShooterTest : MonoBehaviour
+{
+    private AltUnityDriver altUnityDriver = new AltUnityDriver();
+
```
Comment (spirit126) [inline Assets/Editor/Tests/ShooterTest.cs:9]: same as above

**Review by spirit126 — CHANGES_REQUESTED** [review]

**Comment by spirit126** [comment]
- Player Prefab is outdated (the new version is in the Main Scene)
- The New Game button does not work anymore

**Comment by SomethingWittyWithATechAspect** [comment]
> * Player Prefab is outdated (the new version is in the Main Scene)
> * The New Game button does not work anymore

done
done

**Review by spirit126 — APPROVED** [review]
tested, fine now

**Pull request merged**